src/test/java/xmlobject/detailed/ExtraNameSpaceTest.java

```diff
@@ -0,0 +1,71 @@
+ *  limitations under the License.
+ */
+package xmlobject.detailed;
+
+import java.io.ByteArrayOutputStream;
+import java.nio.charset.StandardCharsets;
+import java.util.HashMap;
+import java.util.Map;
+
+import org.apache.xmlbeans.XmlObject;
+import org.apache.xmlbeans.XmlOptions;
+
+import javax.xml.parsers.DocumentBuilder;
+import javax.xml.parsers.DocumentBuilderFactory;
+
+import org.junit.jupiter.api.Test;
+import org.w3c.dom.Document;
+import org.w3c.dom.Element;
+import static org.junit.jupiter.api.Assertions.assertEquals;
+
+public class ExtraNameSpaceTest {
+    // Test whether xmlobject can write extraNamespaces
+    @Test
+    public void testSaveExtraNamespaces() throws Exception {
+        // Creating and setting up a namespace
+        Map<String, String> extraNamespaces = new HashMap<>();
+        extraNamespaces.put("uof", "http://www.nits.org.cn/uof3/zh-cn/2021/uof");
+        extraNamespaces.put("图", "http://www.nits.org.cn/uof3/zh-cn/2021/graph");
+        extraNamespaces.put("字", "http://www.nits.org.cn/uof3/zh-cn/2021/wordproc");
+        extraNamespaces.put("表", "http://www.nits.org.cn/uof3/zh-cn/2021/spreadsheet");
+        extraNamespaces.put("演", "http://www.nits.org.cn/uof3/zh-cn/2021/presentation");
+        
+        XmlOptions xmlOptions = new XmlOptions();
+        xmlOptions.setSaveExtraNamespaces(extraNamespaces);
+
+        assertEquals("http://www.nits.org.cn/uof3/zh-cn/2021/uof", extraNamespaces.get("uof"));
+        assertEquals("http://www.nits.org.cn/uof3/zh-cn/2021/graph", extraNamespaces.get("图"));
+        assertEquals("http://www.nits.org.cn/uof3/zh-cn/2021/wordproc", extraNamespaces.get("字"));
+        assertEquals("http://www.nits.org.cn/uof3/zh-cn/2021/spreadsheet", extraNamespaces.get("表"));
+        assertEquals("http://www.nits.org.cn/uof3/zh-cn/2021/presentation", extraNamespaces.get("演"));
+
+        ByteArrayOutputStream outputStream = new ByteArrayOutputStream();
+
+        XmlObject doc = XmlObject.Factory.parse("<foo><a/></foo>");
+
+        doc.save(outputStream, xmlOptions);
+        String outputXml = outputStream.toString(StandardCharsets.UTF_8);
+        DocumentBuilderFactory factory = DocumentBuilderFactory.newInstance();
+        DocumentBuilder builder = factory.newDocumentBuilder();
+        Document document = builder.parse(new java.io.ByteArrayInputStream(outputXml.getBytes(StandardCharsets.UTF_8)));
+        Element root = document.getDocumentElement();
+
+        assertEquals("http://www.nits.org.cn/uof3/zh-cn/2021/uof", root.getAttribute("xmlns:uof"));
+        assertEquals("http://www.nits.org.cn/uof3/zh-cn/2021/graph", root.getAttribute("xmlns:图"));
+        assertEquals("http://www.nits.org.cn/uof3/zh-cn/2021/wordproc", root.getAttribute("xmlns:字"));
+        assertEquals("http://www.nits.org.cn/uof3/zh-cn/2021/spreadsheet", root.getAttribute("xmlns:表"));
+        assertEquals("http://www.nits.org.cn/uof3/zh-cn/2021/presentation", root.getAttribute("xmlns:演"));

```

Comment:
Not a single one of these namespaces is used in the XML. I am -1 to merging this. I see no point in supporting adding unused namespace declarations.